raccoglitesi.go

```diff
@@ -168,102 +172,123 @@ func getDocenti(codiceDipartimento string) []Docente {
 	return docenti
 }
 
+var replaceRegexForOutput = regexp.MustCompile(`/\s\s+/gi`)
+
 func generateOutput(dip Dipartimento, docenti []Docente) string {
-	output := fmt.Sprintf("= Tesi %s\n:toc:\n", dip.nome)
+	b := strings.Builder{}
+
+	b.WriteString(fmt.Sprintf("= Tesi %s\n:toc:\n", dip.nome))
 
 	for _, docente := range docenti {
-		output += fmt.Sprintf("\n== %s\n%s | %s[sito web]\n", docente.nome, docente.ruolo, docente.url)
+		b.WriteString(fmt.Sprintf("\n== %s\n%s | %s[sito web]\n", docente.nome, docente.ruolo, docente.url))
 
 		for _, sezioneTesi := range docente.tesi {
-			output += fmt.Sprintf("\n=== %s\n", sezioneTesi.titolo)
+			b.WriteString(fmt.Sprintf("\n=== %s\n", sezioneTesi.titolo))
 
 			for _, sottoSezioneTesi := range sezioneTesi.elementi {
-				output += fmt.Sprintf("\n==== %s\n", sottoSezioneTesi.titolo)
+				b.WriteString(fmt.Sprintf("\n==== %s\n", sottoSezioneTesi.titolo))
 
 				for _, nome := range sottoSezioneTesi.elementi {
-
-					output += fmt.Sprintf("* pass:[%s]\n", nome)
+					b.WriteString(fmt.Sprintf("* pass:[%s]\n", nome))
 				}
-
 			}
 		}
 	}
 
-	output = regexp.MustCompile(`/\s\s+/gi`).ReplaceAllString(output, " ")
+	output := b.String()
+	output = replaceRegexForOutput.ReplaceAllString(output, " ")
 	return output
 }
 
-func saveOutput(dip Dipartimento, output string) string {
+func saveOutput(dip Dipartimento, output string) (string, error) {
 	fileName := fmt.Sprintf("%s.adoc", dip.code)
 	filePath := path.Join(*dirName, fileName)
 
-	os.MkdirAll(*dirName, os.ModePerm)
-
-	file, err := os.Create(filePath)
-	defer file.Close()
+	err := os.MkdirAll(*dirName, os.ModePerm)
 	if err != nil {
-		log.Fatal(err)
+		return "", fmt.Errorf("impossibile creare la directory di output: %s", err)
 	}
 
-	_, err = file.WriteString(output)
+	err = os.WriteFile(filePath, []byte(output), 0666)
 	if err != nil {
-		log.Fatal(err)
+		return "nil", fmt.Errorf("impossibile salvare il file di output: %s", err)
 	}
 
-	return filePath
+	return filePath, nil
 }
 
 func mostraListaDipartimenti(dipartimenti []Dipartimento) {
-	log.Println(
-		"Questa e' la lista dei dipartimenti da cui pui scegliere!\n    Inserire il campo `Codice`!")
+	fmt.Println("Questa e' la lista dei dipartimenti da cui pui scegliere:")
```

Comment:
Era già così, ma possiamo usare l'accento anziché l'apostrofo se non vogliamo tradurre in inglese.